### ISSUE TITLE
Koin Compiler Plugin 1.1.0 🎉

### ISSUE THREAD

**Issue by arnaudgiuliani** [InsertKoinIO/koin-compiler-plugin#82]
Hey there, just to share about 1.1.0 release, A compile-safety-focused release.

**Fewer false positives in multi-module apps.** Compile-time dependency validation now
checks your full assembled Koin graph at the real entry point (`startKoin` /
`koinApplication` / `@KoinApplication`), instead of validating each module on its own. This
removes a class of false "missing dependency" errors that could fire even when your app's
dependency graph was completely correct.

**`allWarningsAsErrors` / `-Werror` friendlier.** New `logSeverity` / `versionCheckSeverity`
options let you silence informational plugin output without hiding real diagnostics.

**Fixed a file-name-length crash** some users hit on macOS with long package/class names.

**Clearer `KOIN-D001` errors** — missing-dependency messages now point to the exact file,
line, and module involved.

⚠️ **One trade-off worth knowing**: a module with no Koin entry point in its own
compilation (most feature/library modules) no longer gets its own compile-time checks — the
graph is still fully verified, just at the real app's entry point instead of piecemeal per
module. Details in the release notes.

Full changelog: https://github.com/InsertKoinIO/koin-compiler-plugin/releases/tag/1.1.0

Feedback and bug reports always welcome 🙏

**Comment by simonvassili-qm** [comment]
Thanks :) 

Do you have a timeframe for kotlin 2.4.10 support?

**Comment by arnaudgiuliani** [comment]
Do you have issues in 2.4.10?

**Comment by Faltenreich** [comment]
Maybe he is talking about the following warning which occurs when building a project with Koin 4.2.2, Koin Compiler Plugin 1.1.0 and Kotlin 2.4.10:

```
Koin compiler plugin: Kotlin 2.4.10 is newer than the newest tested version (2.4.0) — proceeding with the 2.4.0 adapter. If compilation fails, check for a koin-compiler-plugin update. Supported versions: 2.3.20, 2.4.0.
```

**Comment by kfaraj** [comment]
@arnaudgiuliani I just updated Koin Compiler Plugin 1.1.0 and I still reproduce the following issue with multi-module architecture:

```kotlin
[Koin][KOIN-D001] Missing dependency: com.kfaraj.samples.koin.multimodule.data.MainRepository
  required by: com.kfaraj.samples.koin.multimodule.feature.MainViewModel (parameter 'repository')
  in module: MainApplication (startKoin)
```

- `:app` provides `MainApplication` and `@Module(includes = [FeatureModule::class]) AppModule`
- `:feature` provides `MainViewModel` and `@Module(includes = [DataModule::class]) FeatureModule`
- `:data` provides `MainRepository` and `@Module DataModule`

```mermaid
flowchart TB
    :app -- implementation --> :feature
    :feature -- implementation --> :data
```

https://github.com/kfaraj/samples/tree/test/koin/koin-multimodule

**Comment by arnaudgiuliani** [comment]
> Maybe he is talking about the following warning which occurs when building a project with Koin 4.2.2, Koin Compiler Plugin 1.1.0 and Kotlin 2.4.10:
> 
> ```
> Koin compiler plugin: Kotlin 2.4.10 is newer than the newest tested version (2.4.0) — proceeding with the 2.4.0 adapter. If compilation fails, check for a koin-compiler-plugin update. Supported versions: 2.3.20, 2.4.0.
> ```

Yep 👍 Will fix it

**Comment by arnaudgiuliani** [comment]
> [@arnaudgiuliani](https://github.com/arnaudgiuliani) I just updated Koin Compiler Plugin 1.1.0 and I still reproduce the following issue with multi-module architecture:
> 
> [Koin][KOIN-D001] Missing dependency: com.kfaraj.samples.koin.multimodule.data.MainRepository
>   required by: com.kfaraj.samples.koin.multimodule.feature.MainViewModel (parameter 'repository')
>   in module: MainApplication (startKoin)
> * `:app` provides `MainApplication` and `@Module(includes = [FeatureModule::class]) AppModule`
> * `:feature` provides `MainViewModel` and `@Module(includes = [DataModule::class]) FeatureModule`
> * `:data` provides `MainRepository` and `@Module DataModule`
> 
> 
> https://github.com/kfaraj/samples/tree/test/koin/koin-multimodule

Will check it 👍

**Comment by arnaudgiuliani** [comment]
@kfaraj  Please have a look at KCP 1.2.1, it should really help on solving A3 full graph 👍
